controllers/component_build_controller.go

```diff
@@ -152,17 +160,66 @@ func (r *ComponentBuildReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 
 	if val, ok := component.Annotations[gitops.PaCAnnotation]; ok && val == "1" {
 		// Use pipelines as code build
-		r.Log.Info("PaC enabled")
-		if err := r.EnsurePACRepository(ctx, component); err != nil {
+		r.Log.Info("Pipelines as Code enabled")
+
+		// Obtain Pipelines as Code callback URL
+		webhookTargetUrl, err := r.getPaCRoutePublicUrl(ctx)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+
+		gitProvider, err := gitops.GetGitProvider(component)
+		if err != nil {
+			r.Log.Error(err, "error detecting git provider")
+			// Do not reconcile, because configuration must be fixed before it is possible to proceed.
+			return ctrl.Result{}, nil
+		}
+
+		// Expect that the secret contains token for Pipelines as Code webhook configuration,
+		// but under <git-provider>.token field. For example: github.token
+		// Also it can contain github-private-key and github-application-id
+		// in case GitHub Application is used instead of webhook.
+		pacSecret := corev1.Secret{}
+		if err := r.NonCachingClient.Get(ctx, types.NamespacedName{Namespace: pipelinesAsCodeNamespace, Name: gitopsprepare.PipelinesAsCodeSecretName}, &pacSecret); err != nil {
+			r.Log.Error(err, "failed to get Pipelines as Code secret")
```

Comment:
an event in `pipelineAsCodeNamespace` seems warranted here as well.

Reply:
ok